### DISCUSSION TITLE
Run jbang-maven-plugin based on quarkus app with resolved config properties

### DISCUSSION THREAD

**Discussion by ennishol** [jbangdev/jbang#1061] [Q&A]
Would it be possible to add support running jbang maven plugin in quarkus application with support of the quarkus profile and custom config sources, so that the script has access to config properties resolved as they would be in quarkus application?

`mvn package  -Dquarkus.config.locations=./src/main/resources/application-custom1.properties -Dquarkus.profile=staging`
or
`mvn package  -Dquarkus.config.locations=./src/main/resources/application-custom2.properties -Dquarkus.profile=production`

```xml
      <plugin>
        <groupId>dev.jbang</groupId>
        <artifactId>jbang-maven-plugin</artifactId>
        <version>0.0.7</version>
        <executions>
          <execution>
            <id>run</id>
            <phase>package</phase>
            <goals>
              <goal>run</goal>
            </goals>
            <configuration>
              <script>src/main/jbang/script.java</script>
              <args>
                <arg>${quarkus.profile}</arg>
                <arg>${quarkus.config.locations}</arg>
              </args>
            </configuration>
          </execution>
        </executions>
      </plugin>
```

**Comment by maxandersen** [comment]
if you need to set system properties you'll want to set it with: <jbangargs> rather than <args> which are argument to your script.java which quarkus does not see.

Does it work then ?

**Comment by ennishol** [comment]
@maxandersen  Sorry, I don't get your comment. My goal is to have the same configuration as the main quarkus application has. Currently, I do parsing of the profile aware properties by myself, but it would be nice, if jbang could provide support for that

**Comment by maxandersen** [comment]
jbang does not really know about quarkus but you can set -D properties via jbangargs which you quarkus script/app then should see.

something like this:

```
  <plugin>
        <groupId>dev.jbang</groupId>
        <artifactId>jbang-maven-plugin</artifactId>
        <version>0.0.7</version>
        <executions>
          <execution>
            <id>run</id>
            <phase>package</phase>
            <goals>
              <goal>run</goal>
            </goals>
            <configuration>
              <script>src/main/jbang/script.java</script>
             <jbangargs>
                <arg>-Dquarkus.profile=${quarkus.profile}</arg>
                <arg>-Dquarkus.config.locations=${quarkus.config.locations}</arg>
              </jbangargs>
              <args>
                <arg>${quarkus.profile}</arg>
                <arg>${quarkus.config.locations}</arg>
              </args>
            </configuration>
          </execution>
        </executions>
      </plugin>
```

**Comment by maxandersen** [comment]
@ennishol I realise you *might* be affected by https://github.com/jbangdev/jbang/issues/1058 - thus please retry ensuring you use latest 0.83.0+  release of jbang and see if it changes things?

**Reply by maxandersen** [reply]
Did it help/fix things @ennishol ?
